### PR TITLE
1.8- sign crash patch

### PR DESCRIPTION
**Pull request by z7087**
https://github.com/CCBlueX/LiquidBounce/pull/2598
didnt test a lot, need more test

### CODE REVIEW

File: src/main/java/ac/grim/grimac/checks/impl/badpackets/BadPacketsU.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package ac.grim.grimac.checks.impl.badpackets;
+
+import ac.grim.grimac.checks.Check;
+import ac.grim.grimac.checks.CheckData;
+import ac.grim.grimac.checks.type.PacketCheck;
+import ac.grim.grimac.player.GrimPlayer;
+import com.github.retrooper.packetevents.event.PacketReceiveEvent;
+
+// See: CrashI
```
Comment (ManInMyVan) [inline src/main/java/ac/grim/grimac/checks/impl/badpackets/BadPacketsU.java]: Use javadoc

```java
import ac.grim.grimac.checks.impl.crash.CrashI;

/**
 * @see CrashI
 */

File: src/main/java/ac/grim/grimac/checks/impl/crash/CrashI.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                    if (!flaggedColorSign && line.indexOf(COLOR_CHAR) != -1) {
+                        player.checkManager.getPacketCheck(BadPacketsU.class).handleColorSign(event);
+                        if (event.isCancelled())
+                            break;
+                        flaggedColorSign = true;
+                    }
+                }
+            }
+            // Will some 384 length strings make stack overflow?
+            catch (StackOverflowError | Exception e) {
```
Comment (SamB440) [inline src/main/java/ac/grim/grimac/checks/impl/crash/CrashI.java:94]: Catching StackOverflowError is cursed

Comment (z7087) [reply src/main/java/ac/grim/grimac/checks/impl/crash/CrashI.java:94]: will try change to fail fast and remove this after ready

**Comment by ghost** [comment]
nice code bro, good job 👍

**Pull request closed without merging**